From 2f5ad1e40ee2f1ef290b1fcaec191f75f0ae5159 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20R=C3=BCth?= <julian.rueth@fsfe.org>
Date: Thu, 28 Dec 2023 01:31:49 -0600
Subject: [PATCH 1/2] Do not typedef ssize_t

our setup for Windows seems to replace ssize_t with long long

this leads to error: cannot combine with previous 'long long'
declaration specifier when reading the code `typedef long long long
long;`
---
 source/libnormaliz/face_lattice.cpp | 4 ----
 source/libnormaliz/full_cone.cpp    | 4 ----
 source/libnormaliz/matrix.cpp       | 4 ----
 source/libnormaliz/simplex.cpp      | 4 ----
 4 files changed, 16 deletions(-)

diff --git a/source/libnormaliz/face_lattice.cpp b/source/libnormaliz/face_lattice.cpp
index 7060feb..0134af0 100644
--- a/source/libnormaliz/face_lattice.cpp
+++ b/source/libnormaliz/face_lattice.cpp
@@ -25,10 +25,6 @@
 #include "libnormaliz/face_lattice.h"
 #include "libnormaliz/vector_operations.h"
 
-#ifdef _MSC_VER
-typedef long long ssize_t;
-#endif
-
 namespace libnormaliz {
 
 using namespace std;
diff --git a/source/libnormaliz/full_cone.cpp b/source/libnormaliz/full_cone.cpp
index 0fd906b..a0528fb 100644
--- a/source/libnormaliz/full_cone.cpp
+++ b/source/libnormaliz/full_cone.cpp
@@ -45,10 +45,6 @@
 #include "libnormaliz/sublattice_representation.h"
 #include "libnormaliz/offload_handler.h"
 
-#ifdef _MSC_VER
-typedef long long ssize_t;
-#endif
-
 //---------------------------------------------------------------------------
 
 namespace libnormaliz {
diff --git a/source/libnormaliz/matrix.cpp b/source/libnormaliz/matrix.cpp
index b926170..b736a40 100644
--- a/source/libnormaliz/matrix.cpp
+++ b/source/libnormaliz/matrix.cpp
@@ -41,10 +41,6 @@
 #include "flint/fmpz_mat.h"
 #endif
 
-#ifdef _MSC_VER
-typedef long long ssize_t;
-#endif
-
 //---------------------------------------------------------------------------
 
 namespace libnormaliz {
diff --git a/source/libnormaliz/simplex.cpp b/source/libnormaliz/simplex.cpp
index 5895662..1eb576a 100644
--- a/source/libnormaliz/simplex.cpp
+++ b/source/libnormaliz/simplex.cpp
@@ -43,10 +43,6 @@
 
 //---------------------------------------------------------------------------
 
-#ifdef _MSC_VER
-typedef long long ssize_t;
-#endif
-
 namespace libnormaliz {
 using namespace std;
 
-- 
2.43.0

